src/rollup/index.ts

```diff
@@ -468,6 +471,9 @@ export default function rollup(
 			.then(chunks => {
 				timeEnd('BUILD', 1);
 
+				// ensure we only do one optimization pass per build
+				let optimized = false;
```

Comment:
I really do no like how we need that flag but admittedly I do not have a better idea on how to get a good size estimate before we call `generate` for the first time.